src/Sql/dbo/Stored Procedures/Organization_EnableCollectionEnhancements.sql

```diff
@@ -0,0 +1,132 @@
+            FROM (
+                     SELECT C.[Id] AS [CollectionId], T.[OrganizationUserId]
+                     FROM [dbo].[Collection] C
+                         INNER JOIN #TempOrgUser T ON C.[OrganizationId] = T.[OrganizationId]
+                 ) AS source
+            LEFT JOIN [dbo].[CollectionUser] AS target
+                ON target.[CollectionId] = source.[CollectionId] AND target.[OrganizationUserId] = source.[OrganizationUserId]
+            WHERE target.[CollectionId] IS NULL;
+
+            -- Update OrganizationUser to clear AccessAll flag
+            UPDATE OU
+            SET [AccessAll] = 0
+            FROM [dbo].[OrganizationUser] OU
+            INNER JOIN #TempOrgUser T ON OU.[Id] = T.[OrganizationUserId]
+
+            -- Drop the temporary table
+            DROP TABLE #TempOrgUser;
+
+        -- Step 3: Update [dbo].[CollectionUser] with [Manage] = 1 for all users with Manager role or 'EditAssignedCollections' permission
+            UPDATE cu
+            SET cu.[ReadOnly] = 0,
+                cu.[HidePasswords] = 0,
+                cu.[Manage] = 1
+            FROM [dbo].[CollectionUser] cu
+            INNER JOIN [dbo].[OrganizationUser] ou
+                ON cu.[OrganizationUserId] = ou.[Id]
+            WHERE ou.[OrganizationId] = @OrganizationId
+              AND (ou.[Type] = 3
+               OR (ou.[Permissions] IS NOT NULL AND ISJSON(ou.[Permissions]) > 0 AND JSON_VALUE(ou.[Permissions], '$.editAssignedCollections') = 'true'))
+
+        -- Step 4: Insert rows to [dbo].[CollectionUser] for Managers and users with 'EditAssignedCollections' permission assigned to groups with collection access
+            INSERT INTO [dbo].[CollectionUser] ([CollectionId], [OrganizationUserId], [ReadOnly], [HidePasswords], [Manage])
+            SELECT cg.[CollectionId], ou.[Id], 0, 0, 1
+            FROM [dbo].[CollectionGroup] cg
+            INNER JOIN [dbo].[GroupUser] gu
+                ON cg.GroupId = gu.GroupId
+            INNER JOIN [dbo].[OrganizationUser] ou
+                ON gu.OrganizationUserId = ou.[Id]
+            WHERE ou.[OrganizationId] = @OrganizationId
+                AND (ou.[Type] = 3
+                    OR (ou.[Permissions] IS NOT NULL AND ISJSON(ou.[Permissions]) > 0 AND JSON_VALUE(ou.[Permissions], '$.editAssignedCollections') = 'true'))
+                AND NOT EXISTS (
+                    SELECT 1 FROM [dbo].[CollectionUser] cu
+                    WHERE cu.[CollectionId] = cg.[CollectionId] AND cu.[OrganizationUserId] = ou.[Id]
+                )
+
+        -- Step 5: Set all Managers to Users
+            UPDATE [dbo].[OrganizationUser]
+            SET [Type] = 2  -- User
+            WHERE [OrganizationId] = @OrganizationId
+              AND [Type] = 3; -- Manager
+
+        -- Step 6: Bump the account revision dates for all users in the organization
+        EXEC [dbo].[User_BumpAccountRevisionDateByOrganizationId] @OrganizationId
```

Comment:
Not all users will be affected by these changes. I think you'd want to use `ByCollectionIds` or `ByOrganizationUserId` to target only those users we've actually updated.

Reply:
I have updated the sproc to bump the dates by `OrganizationUserId`